tests/regression/task_spec.py

```diff
@@ -40,10 +41,43 @@ class TaskSpec:
     Contains a list of functions to be compared with DiffKemp during the test.
     """
     def __init__(self, spec, task_name, kernel_path):
-        self.old_kernel_dir = os.path.join(kernel_path, spec["old_kernel"])
-        self.new_kernel_dir = os.path.join(kernel_path, spec["new_kernel"])
         self.name = task_name
         self.task_dir = os.path.join(tasks_path, task_name)
+        # Paths to test data directories (test_data/<taskname>/{old,new})
+        self.old_source_dir = os.path.join(self.task_dir, "old")
+        self.new_source_dir = os.path.join(self.task_dir, "new")
+
+        # If paths to the real kernel sources exist, create corresponding
+        # KernelSourceTree objects
+        self.old_kernel_dir = os.path.join(kernel_path, spec["old_kernel"])
+        self.new_kernel_dir = os.path.join(kernel_path, spec["new_kernel"])
+        self.old_kernel_source_tree = None
+        self.new_kernel_source_tree = None
+        if os.path.isdir(self.old_kernel_dir):
+            self.old_kernel_source_tree = KernelSourceTree(
+                self.old_kernel_dir,
+                KernelLlvmSourceBuilder(self.old_kernel_dir))
+        if os.path.isdir(self.new_kernel_dir):
+            self.new_kernel_source_tree = KernelSourceTree(
+                self.new_kernel_dir,
+                KernelLlvmSourceBuilder(self.new_kernel_dir))
```

Comment:
I am not sure what is going to happen if kernel directories do not exists. I am not sure if I understand correctly, but it would probably fail around [here](https://github.com/viktormalik/diffkemp_/blob/2c876c431fde337176d411df493ffa17db54d044/tests/regression/mock_source_tree.py#L41), right? Wouldn't it be better to fail around here? It is possible to raise and exception in tests.

Reply:
I don't think that we can fail here. Non-existing kernel directory is a genuine option, the only problem would be when the kernel directory doesn't exist *and* the files are not cached. In order to fail here, we'd have to detect that the cached files are missing and that's hard to do here as we don't know what files to look for (it depends on the task type).

In the current implementation, the failure will occur in `KernelSourceTree` (or `KernelLlvmSourceBuilder`) when trying to find the LLVM IR module for a given function/module/sysctl.

Reply:
I think I understand now, that I have revisited it with clarification. I through that when `self.old_kernel_source_tree` is `None`, then it is going to fail, but there is more logic behind it.